atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/StreamStoreRenderer.java

```diff
@@ -558,52 +561,86 @@ private void unmarkStreamsAsUsed() {
             }
 
             private void storeBlocksAndGetFinalMetadata() {
+                String compression = clientSideCompression.compressionType;
```

Comment:
I would propose here that you instead add a method to your compression enum which is something like `InputStream compress(InputStream input)` which does the wrapping for you. In this way you can do any extra exception handling that you need there.